### ISSUE TITLE
Tooltips are positioned too close to triggering element

### ISSUE THREAD

**Issue by cybrox** [ember-bootstrap/ember-bootstrap#2118]
Thank you for maintaining Ember-Bootstrap!

We are currently in the process of upgrading a legacy Ember application and have migrated everything to Ember 4.12 LTS. One of the issues we have encountered is that the tooltips are a lot closer to the triggering element than in Bootstrap normally. This causes the arrow to overlap the element, which negatively impacts the hover/click detection, especially on small words or buttons.

This is a screenshot from the [Bootstrap docs](https://getbootstrap.com/docs/4.0/components/tooltips/) showing the tooltip positioned correctly.
![2024-03-28_08-36-17](https://github.com/ember-bootstrap/ember-bootstrap/assets/2383736/1258bb89-cc0c-4e67-9920-f95571f708b1)

This is a screenshot from the [Ember-Bootstrap docs](https://www.ember-bootstrap.com/components/tooltip/) showing the tooltip positioned incorrectly.
![2024-03-27_17-31-43](https://github.com/ember-bootstrap/ember-bootstrap/assets/2383736/892ef7c6-dca2-4a02-94f1-8394c35adb51)

This is a screenshot of an example UI with a tooltip on a button. It looks like the generated `transform` directive does not take into account the size of the arrow, so the arrow overlaps the triggering button element.
![2024-03-28_08-39-48](https://github.com/ember-bootstrap/ember-bootstrap/assets/2383736/429d1571-52ab-4544-acac-6e55370ea012)

---

Output from `ember bootstrap:info`:
```
Npm packages:
ember-bootstrap: ^6.2.0 -> 6.2.0
ember-cli: ^4.12.1 -> 4.12.2
bootstrap: ^5.0.0 -> 5.3.3
bootstrap-sass: n/a
ember-cli-sass: ^11.0.1 -> 11.0.1
ember-cli-less: n/a
Bower packages:
bootstrap: n/a
bootstrap-sass: n/a
ember-bootstrap configuration:
bootstrapVersion: 5
importBootstrapCSS: false
insertEmberWormholeElementToDom: false
```

Output from `ember --version`
```
ember-cli: 4.12.2
node: 18.16.1
os: linux x64
```

**Comment by jelhan** [comment]
What version of ember-popper-modifier and ember-modifier do you actually use?

Can you share a reproduction?

**Comment by cybrox** [comment]
Hi @jelhan, thanks for the reply!

I have created [a repository](https://github.com/cybrox/ember-bootstrap-tooltip-issue) that is a minimal reproduction of the issue. The behaviour is consistent with the [ember-bootstrap website](https://www.ember-bootstrap.com/components/tooltip), which also exhibits the issue.

I have not explicitly added `ember-popper-modifier` or `ember-modifier` to my project but according to the lockfile, it has pulled the following versions:
* `ember-popper-modifier`: `4.1.0`
* `ember-modifier`: `4.1.0`

When running the reproduction library above with `ember run start`, it has the same tooltip positioning issue:

![2024-04-02_10-32-54](https://github.com/ember-bootstrap/ember-bootstrap/assets/2383736/b3470c27-a155-4c5b-a9bd-d7bf2a2322a8)

**Comment by jelhan** [comment]
Thanks a lot for providing the reproduction. Also good to know that it affects the documentation website as well. Sounds as if it is a general regression.

I'm not sure how soon I (or another maintainer) have time looking into it. If you could trace it down to a specific version, which broke it, that would be very helpful. It might be caused by a change in `ember-bootstrap` itself or in `ember-popper-modifier`.

**Comment by cybrox** [comment]
Thanks! I'm not too familiar with the interaction between these packages, so I just did some "logical" debugging.   
This might give some insight. (`cfg` is the `bootstrapVersion` configured in `ember-cli-build.js`):

I started lowering the version of all packages and the issue stayed persistent:
`bootstrap` | `ember-modifier` | `ember-popper-modifier` | `ember-bootstrap` | cfg | result
--- | --- | --- | --- | --- | --- |
`^5.3.3` | `3.0.0` | `3.0.0` | `6.2.0` | 5 | ❌ Broken as described
`^5.3.3` | `3.0.0` | `3.0.0` | `5.1.1` | 5 | ❌ Broken as described
`^5.3.3` | `2.1.2` | `3.0.0` | `5.0.0` | 5 | ❌ Broken as described
`^5.3.3` | `2.1.2` | `2.0.1` | `5.0.0` | 5 | ❌ Broken as described

I then looked up the last working state on my project and started increasing the version of all packages from there. The notable difference is, that my project still used Bootstrap 4 before we upgraded.

`bootstrap` | `ember-modifier` | `ember-popper-modifier` | `ember-bootstrap` | cfg | result
--- | --- | --- | --- | --- | --- |
`^4.6.2` | `3.1.0` | `2.0.0` | `5.1.1` | 4 | ✅ Works as expected
`^4.6.2` | `4.1.0` | `4.1.0` | `6.2.0` | 4 | ✅ Works as expected
`^4.6.2` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌❌ Very broken*
`^5.0.0` | `4.1.0` | `4.1.0` | `6.2.0` | 4 | ❌ Broken as described
`^5.0.0` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌ Broken as described
`^5.3.3` | `4.1.0` | `4.1.0` | `6.2.0` | 4 | ❌❌ Very broken**
`^5.3.3` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌ Broken as described

_* = no arrows, positioning inconsistent between vertical and horizontal_   
_** = no arrows_

**To summarize:**
* I don't think `ember-modifier` or `ember-popper-modifier` versions have an impact here.
* Up until `bootstrap<=4.6.2`, the tooltips work fine when using Bootstrap 4 (but not Bootstrap 5) even with the latest version of all other (and this) library.
* Starting from `bootstrap=5.0.0`, the tooltip exhibit the described issue using both Bootstrap 4 and 5
* Using the latest version of `bootstrap=5.3.3`, the tooltip does not have an arrow when using Bootstrap 4 at all.

When I created the reproduction project, I ran `ember generate ember-bootstrap`, which generated me an `ember-cli-build.js` entry which still manually imports the styles despite this causing a deprecation warning, so I disabled `importBootstrapCSS` and instead added `app.import('node_modules/bootstrap/dist/css/bootstrap.css');` but this does not seem to have any impact.

**To summa-summarize:** It looks like the issue is entirely dependent on the version of the [`bootstrap`](https://www.npmjs.com/package/bootstrap?activeTab=versions) package. I didn't find any combination where Bootstrap 5 works properly but maybe I missed it or this issue was simply overlooked during testing as it's not very obvious at first sight.

**Comment by jelhan** [comment]
We may want to have a look which options Bootstrap provides to Popper for the tooltip. I feel porting over what Bootstrap does is the safest option in such cases.

**Comment by cybrox** [comment]
Upon reading the discussion in Discord, I double-checked the test setup and noticed I forgot to remove the caret (`^`) in the Bootstrap version in the tests above.

When pinning `bootstrap` to an exact version, I get the following results:

`bootstrap` | `ember-modifier` | `ember-popper-modifier` | `ember-bootstrap` | cfg | result
--- | --- | --- | --- | --- | --- |
`5.3.3` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌ Broken as described
`5.3.0` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌ Broken as described
`5.3.0-alpha3` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ❌ Broken as described
`5.3.0-alpha2` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.3.0-alpha1` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.2.3` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.2.2` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.2.0` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.1.0` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected
`5.0.0` | `4.1.0` | `4.1.0` | `6.2.0` | 5 | ✅ Works as expected

So the regression happened with the [`5.3.0-alpha3` release](https://github.com/twbs/bootstrap/releases/tag/v5.3.0-alpha3) of the `bootstrap` package.

**Comment by SanderKnauff** [comment]
This has been addressed in #2119.
Should we create a new release specifically for this bugfix?

**Comment by jelhan** [comment]
Thanks a lot for fixing. I can quickly release it.

**Comment by jelhan** [comment]
Released as [v6.3.0](https://github.com/ember-bootstrap/ember-bootstrap/releases/tag/v6.3.0).
